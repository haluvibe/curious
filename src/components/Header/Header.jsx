import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import classNames from "classnames";
import Drawer from "material-ui/Drawer";
import AppBar from "material-ui/AppBar";
import Toolbar from "material-ui/Toolbar";
import List from "material-ui/List";
import Typography from "material-ui/Typography";
import Divider from "material-ui/Divider";
import IconButton from "material-ui/IconButton";
import MenuIcon from "material-ui-icons/Menu";
import ChevronLeftIcon from "material-ui-icons/ChevronLeft";
import ChevronRightIcon from "material-ui-icons/ChevronRight";
import { ListItem, ListItemIcon, ListItemText } from "material-ui/List";
import ReportIcon from "material-ui-icons/Report";
import StarIcon from "material-ui-icons/Star";
import Button from "material-ui/Button";

const drawerWidth = 240;

const styles = theme => ({
  appBar: {
    position: "absolute",
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    })
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    }),
    "& $loginButton": {
      marginLeft: 12,
      marginRight: 0
    }
  },
  flex: {
    flex: 1
  },
  loginButton: {
    marginLeft: 12,
    marginRight: 24
  },
  menuButton: {
    marginLeft: 12,
    marginRight: 36
  },
  hide: {
    display: "none"
  },
  drawerPaper: {
    position: "relative",
    height: "100vh",
    width: drawerWidth,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  drawerPaperClose: {
    width: 60,
    overflowX: "hidden",
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    })
  },
  drawerInner: {
    // Make the items inside not wrap when transitioning:
    width: drawerWidth
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    ...theme.mixins.toolbar
  },
  content: {
    width: "100%",
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    padding: 24,
    height: "calc(100% - 56px)",
    marginTop: 56,
    [theme.breakpoints.up("sm")]: {
      height: "calc(100% - 64px)",
      marginTop: 64
    }
  }
});

class MiniDrawer extends React.Component {
  state = {
    open: false
  };

  handleDrawerOpen = () => {
    this.setState({ open: true });
  };

  handleDrawerClose = () => {
    this.setState({ open: false });
  };

  render () {
    const { classes, theme } = this.props;

    return (
      <Fragment>
        <AppBar
          className={classNames(
            classes.appBar,
            this.state.open && classes.appBarShift
          )}
        >
          <Toolbar disableGutters={!this.state.open}>
            <IconButton
              color="inherit"
              aria-label="open drawer"
              onClick={this.handleDrawerOpen}
              className={classNames(
                classes.menuButton,
                this.state.open && classes.hide
              )}
            >
              <MenuIcon />
            </IconButton>
            <Typography
              variant="title"
              color="inherit"
              noWrap
              className={classes.flex}
            >
              Curious
            </Typography>
            <Button className={classes.loginButton} color="inherit">
              Login
            </Button>
          </Toolbar>
        </AppBar>
        <Drawer
          variant="permanent"
          classes={{
            paper: classNames(
              classes.drawerPaper,
              !this.state.open && classes.drawerPaperClose
            )
          }}
          open={this.state.open}
        >
          <div className={classes.drawerInner}>
            <div className={classes.drawerHeader}>
              <IconButton onClick={this.handleDrawerClose}>
                {theme.direction === "rtl" ?
                  <ChevronRightIcon />
                  :
                  <ChevronLeftIcon />
                }
              </IconButton>
            </div>
            <Divider />
            <List>
              <ListItem button>
                <ListItemIcon>
                  <ReportIcon />
                </ListItemIcon>
                <ListItemText primary="Inbox" />
              </ListItem>
              <ListItem button>
                <ListItemIcon>
                  <StarIcon />
                </ListItemIcon>
                <ListItemText primary="Starred" />
              </ListItem>
            </List>
          </div>
        </Drawer>
      </Fragment>
    );
  }
}

MiniDrawer.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
};

export default withStyles(styles, { withTheme: true })(MiniDrawer);
