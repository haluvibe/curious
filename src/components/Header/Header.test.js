import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Header from "./Header";

describe("<Header />", () => {
  const component = shallow(<Header />).dive();
  it("renders correctly", () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  describe("when the open and close menu buttons are clicked", () => {
    it("handles openMenu click correctly", () => {
      component
        .find("WithStyles(IconButton)")
        .first()
        .simulate("click");
      expect(component.state().open).toBe(true);
    });

    it("handles closesMenu click correctly", () => {
      component
        .find("WithStyles(IconButton)")
        .at(1)
        .simulate("click");
      expect(component.state().open).toBe(false);
    });
  });
});
