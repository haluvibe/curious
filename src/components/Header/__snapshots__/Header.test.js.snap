// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<React.Fragment>
  <WithStyles(AppBar)
    className="MiniDrawer-appBar-1"
  >
    <WithStyles(Toolbar)
      disableGutters={true}
    >
      <WithStyles(IconButton)
        aria-label="open drawer"
        className="MiniDrawer-menuButton-5"
        color="inherit"
        onClick={[Function]}
      >
        <pure(Menu) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        className="MiniDrawer-flex-3"
        color="inherit"
        noWrap={true}
        variant="title"
      >
        Curious
      </WithStyles(Typography)>
      <WithStyles(Button)
        className="MiniDrawer-loginButton-4"
        color="inherit"
      >
        Login
      </WithStyles(Button)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    classes={
      Object {
        "paper": "MiniDrawer-drawerPaper-7 MiniDrawer-drawerPaperClose-8",
      }
    }
    open={false}
    variant="permanent"
  >
    <div
      className="MiniDrawer-drawerInner-9"
    >
      <div
        className="MiniDrawer-drawerHeader-10"
      >
        <WithStyles(IconButton)
          onClick={[Function]}
        >
          <pure(ChevronLeft) />
        </WithStyles(IconButton)>
      </div>
      <WithStyles(Divider) />
      <WithStyles(List)>
        <WithStyles(ListItem)
          button={true}
        >
          <WithStyles(ListItemIcon)>
            <pure(Report) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            primary="Inbox"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          button={true}
        >
          <WithStyles(ListItemIcon)>
            <pure(Star) />
          </WithStyles(ListItemIcon)>
          <WithStyles(ListItemText)
            primary="Starred"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </div>
  </WithStyles(Drawer)>
</React.Fragment>
`;
